monai/data/meta_tensor.py

```diff
@@ -523,10 +523,10 @@ def ensure_torch_and_prune_meta(
             By default, a `MetaTensor` is returned.
             However, if `get_track_meta()` is `False`, a `torch.Tensor` is returned.
         """
-        img = convert_to_tensor(im)  # potentially ascontiguousarray
+        img = convert_to_tensor(im, track_meta=get_track_meta() and meta is not None)  # potentially ascontiguousarray
```

Comment:
Hi @wyli, I'm a little confused about this line, if im is a MetaTensor, `get_track_meta()` is `True` and meta is `None`, then the meta information will be lost. Is that right?

Reply:
Yes, setting meta=None essentially skips the metatensor creation. In that case it'll return a torch tensor.

Reply:
Thanks for the explanation, but the case I considering is that when input is already `MetaTensor`, then I can't preserve the affine information unless I reassign it in the meta dict. Maybe it can be `track_meta=get_track_meta() or meta is not None`? Or I misunderstood it?
```
inp = MetaTensor([1,2,3], meta={'affine': affine, 'space': "LAS"})
out = MetaTensor.ensure_torch_and_prune_meta(inp, meta=None) --> affine will be lost
```

Reply:
that's the current `dev` branch's behaviour as well, I didn't try to change it.  I think it makes sense to use `None` to indicate no meta (regular tensor), and `{}` to indicate an empty meta (default affine)

Reply:
Ok, got it, thanks!

Reply:
thanks, I updated the docstring to make it clear.